python_modules/dagster/dagster_tests/core_tests/asset_defs_tests/test_partitioned_assets.py

```diff
@@ -439,3 +441,125 @@ def load_input(self, context):
+        },
+        partitions_def=partitions_def,
+    )
+    def upstream_asset():
+        pass
+
+    @asset(partitions_def=partitions_def)
+    def downstream_asset_1(upstream_asset_1: int):
+        pass
+
+    @asset(partitions_def=partitions_def)
+    def downstream_asset_2(upstream_asset_2: int):
+        pass
+
+    assert get_upstream_partitions_for_partition_range(
+        downstream_asset_1,
+        upstream_asset,
+        AssetKey("upstream_asset_1"),
+        PartitionKeyRange("a", "c"),
+    ) == PartitionKeyRange("a", "c")
+
+    assert get_upstream_partitions_for_partition_range(
+        downstream_asset_2,
+        upstream_asset,
+        AssetKey("upstream_asset_2"),
+        PartitionKeyRange("a", "c"),
+    ) == PartitionKeyRange("a", "c")
+
+    assert get_downstream_partitions_for_partition_range(
+        downstream_asset_1,
+        upstream_asset,
+        AssetKey("upstream_asset_1"),
+        PartitionKeyRange("a", "c"),
+    ) == PartitionKeyRange("a", "c")
+
+    assert get_downstream_partitions_for_partition_range(
+        downstream_asset_2,
+        upstream_asset,
+        AssetKey("upstream_asset_2"),
+        PartitionKeyRange("a", "c"),
+    ) == PartitionKeyRange("a", "c")
+
+
+def test_single_partitioned_multi_asset_job():
+    partitions_def = StaticPartitionsDefinition(["a", "b", "c", "d"])
+
+    class MyIOManager(IOManager):
+        def handle_output(self, context, obj):
+            assert context.asset_partition_key == "b"
+
+        def load_input(self, context):
+            assert False, "shouldn't get here"
+
+    @multi_asset(
+        outs={
+            "out1": Out(asset_key=AssetKey("my_asset_1")),
+            "out2": Out(asset_key=AssetKey("my_asset_2")),
+        },
+        partitions_def=partitions_def,
+    )
+    def my_asset():
+        return (Output(1, output_name="out1"), Output(2, output_name="out2"))
+
+    my_job = build_assets_job(
+        "my_job",
+        assets=[my_asset],
+        resource_defs={"io_manager": IOManagerDefinition.hardcoded_io_manager(MyIOManager())},
+    )
+    result = my_job.execute_in_process(partition_key="b")
+    assert result.asset_materializations_for_node("my_asset") == [
+        AssetMaterialization(asset_key=AssetKey(["my_asset_1"]), partition="b"),
+        AssetMaterialization(asset_key=AssetKey(["my_asset_2"]), partition="b"),
+    ]
+
+
+def test_two_partitioned_multi_assets_job():
+    partitions_def = StaticPartitionsDefinition(["a", "b", "c", "d"])
+
+    @multi_asset(
+        outs={
+            "out1": Out(asset_key=AssetKey("upstream_asset_1")),
+            "out2": Out(asset_key=AssetKey("upstream_asset_2")),
+        },
+        partitions_def=partitions_def,
+    )
+    def upstream_asset():
+        return (Output(1, output_name="out1"), Output(2, output_name="out2"))
+
+    @asset(partitions_def=partitions_def)
+    def downstream_asset_1(upstream_asset_1: int):
+        pass
```

Comment:
The tests are complaining that `upstream_asset_1` isn't used (same with the other assets).  You could replace `pass` with `del upstream_asset_1` to address this.